rosbag2_cpp/src/rosbag2_cpp/info.cpp

```diff
@@ -112,10 +147,19 @@ std::vector<std::shared_ptr<rosbag2_service_info_t>> Info::read_service_info(
     while (storage->has_next()) {
       auto bag_msg = storage->read_next();
 
-      // Check if topic is service event topic
-      auto one_service_info = all_service_info.find(bag_msg->topic_name);
-      if (one_service_info == all_service_info.end()) {
-        continue;
+      auto action_topic_type = get_action_topic_type_from_topic_name(bag_msg->topic_name);
```

Comment:
Suggesting to rename `action_topic_type` to the `action_interface_type`.
Also, strongly recommend to speedup check by creating a separate hash map for topic_name to action name or using full topic_name as a key in the `all_action_info`. The latter option is more preferable. 

In addition, please correct me if I am wrong, but I think that in the current design, we have a bug or incorrect behavior when we are using `action_name` instead of the full topic name as a key in the `all_action_info`.

Reply:
> Suggesting to rename action_topic_type to the action_interface_type.
> Also, strongly recommend to speedup check by creating a separate hash map for topic_name to action name or using full topic_name as a key in the all_action_info. The latter option is more preferable.

Okay. 

> In addition, please correct me if I am wrong, but I think that in the current design, we have a bug or incorrect behavior when we are using action_name instead of the full topic name as a key in the all_action_info.

Using the action name is better than using the full topic name (action interface name) for displaying information about each action in action info.